server.js

```diff
@@ -1,35 +1,197 @@
+})
+
+const categorySchema = new mongoose.Schema({
+  description: String
+})
+const Category = mongoose.model('Category', categorySchema)
+
+const Nominee = mongoose.model('Nominee', nomineeSchema)
+
+if (process.env.RESET_D) {
+  const seedDB = async () => {
+    await Nominee.deleteMany()
+    await Category.deleteMany()
+
+    const categoriesArray = []
+
+    categoriesJason.forEach(async (item) => {
+      const category = new Category(item)
+      categoriesArray.push(category)
+      await category.save()
+    })
+
+    await goldenGlobesData.forEach(async (nominee) => {
+      const newNominee = new Nominee({
+        ...nominee,
+        // eslint-disable-next-line max-len
+        category: categoriesArray.find((oneCategory) => oneCategory.description === nominee.category)
+      })
+      await newNominee.save()
+    })
+  }
+  seedDB()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app));
+})
+
+app.get('/categories', async (req, res) => {
+  const categories = await Category.find()
+  res.json(categories)
+})
+
+app.get('/categories/:categoryId/nominees', async (req, res) => {
+  const { categoryId } = req.params
+  const { awardYear, nominee, film, win } = req.query
+  
+  try {
+    const query = {}
+    query.category = categoryId
+    if (awardYear) {
+      query.year_award = awardYear
+    }
+    if (nominee) {
+      query.nominee = { $regex: new RegExp(nominee, "i") }
+    }
+    if (film) {
+      query.film = { $regex: new RegExp(film, "i") }
+    }
+    if (win) {
+      query.win = win
+    }
+    
+    const nomineesCategory = await Nominee.find(query).populate('category')
+    if (nomineesCategory) {
+      res.json(nomineesCategory);
+    } else {
+      res.status(404).json({ error: "Not found" })
+    }
+  } catch (error) {
+    res.status(400).json({ error })
+  }
+});
+
+app.get('/nominees', async (req, res) => {
+  const { awardYear, nominee, page, perPage } = req.query
+  try {
+    const query = {}
+    if (awardYear) {
+      query.year_award = awardYear
+    }
+    if (nominee) {
+      query.nominee = { $regex: new RegExp(nominee, "i") }
+    }
+    const nominees = await Nominee.find(query).populate('category').skip(((Number(page) - 1) * Number(perPage))).limit(Number(perPage))
+    res.json(nominees);
+  } catch (error) {
+    res.status(400).json({ error })
+  }
+})
+
+app.get('/nominees/:id', async (req, res) => {
+  const { id } = req.params
+  try {  
+    const nominee = await Nominee.findById(id)
+    if (nominee) {
+      res.json(nominee)
+    } else {
+      res.status(404).json({ error: "Not found" })
+    }
+  } catch (error) {
+    res.status(400).json({ error })
+  }
+})
+
+app.get('/winners', async (req, res) => {
```

Comment:
This path (/winners) doesn't work for me. I get an error message directly. 

![Skärmavbild 2021-05-19 kl  18 36 27](https://user-images.githubusercontent.com/51661887/118850998-9dcc0000-b8d1-11eb-8493-20fa4db9e8d7.png)

I looked through your code but I can't find what the problem could be. Nonetheless, I had the same problem with my code and it was because of this code part: 

`
 {        $skip: Number((page - 1) * perPage + 1)      },      {        $limit: Number(perPage)      }`

During my 1:1 with Maks, we talked about it and, of course, it worked great when I mentioned it to him ;)  But then when I tried it again it stopped working. So I had to comment out the $limit. I have asked Karol about it in my pull request as well. Anyway, look at it and let me know if it might be the same problem for you.